### PR TITLE
acpica-tools: 20220331 → 20221020

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/acpica-tools/default.nix [hunk 1]
```diff
@@ -27,7 +29,10 @@ stdenv.mkDerivation rec {
     "iasl"
   ];
 
-  NIX_CFLAGS_COMPILE = "-O3";
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-dangling-pointer"
```
Comment (AndersonTorres) [inline pkgs/tools/system/acpica-tools/default.nix:33]: 1. why?
2. failing on Darwin.

Comment (fabianhjr) [reply pkgs/tools/system/acpica-tools/default.nix]: Apologies, extracting some version bumps from https://github.com/NixOS/nixpkgs/pull/206907 and forgot to remove gcc12 specific changes.

Removed that flag addition from this merge request.

File: pkgs/tools/system/acpica-tools/default.nix [hunk 2]
```diff
@@ -7,11 +7,13 @@
...
-  version = "20220331";
+  version = "20221020";
 
   src = fetchurl {
-    url = "https://acpica.org/sites/acpica/files/acpica-unix-${version}.tar.gz";
-    hash = "sha256-rK/2ixTx4IBOu/xLlyaKTMvvz6BTsC7Zkk8rFNipjiE=";
+    # 20221020 has a weird filename published: https://acpica.org/node/201
+    url = "https://acpica.org/sites/acpica/files/acpica-unix-${version}.tar_0.gz";
+    hash = "sha256-M6LjlKygylfUAYr+PaNA361etFsbkwDoHdWV/aB88cU=";
+    name = "acpica-unix-${version}.tar.gz";
```
Comment (AndersonTorres) [inline pkgs/tools/system/acpica-tools/default.nix:16]: Usually `name` attribute comes first when it is used on (sub)expressions.

Comment (fabianhjr) [reply pkgs/tools/system/acpica-tools/default.nix]: Changed

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**